package/samplers/auto_sampler/_sampler.py

```diff
@@ -145,12 +160,35 @@ def _include_conditional_param(self, study: Study) -> bool:
     def _determine_multi_objective_sampler(
         self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
     ) -> BaseSampler:
-        # TODO(nabenabe): Add more efficient heuristic for MO.
-        if isinstance(self._sampler, NSGAIISampler):
-            return self._sampler
-
+        # TODO(nabenabe): Consider any better way to early-return for runtime performance.
         seed = self._rng.rng.randint(_MAXINT32)
-        return NSGAIISampler(constraints_func=self._constraints_func, seed=seed)
+        if self._include_conditional_param(study):
+            if not isinstance(self._sampler, TPESampler):
+                return self._get_tpe_sampler(seed)
```

Comment:
Perhaps, conditional params exist and self._sampler is already TPE, we can simply return here?
```suggestion
                return self._get_tpe_sampler(seed)
            else:
                return self._sampler
```